File: ch05/union_undef.c
#include <stdio.h>
typedef union  {
    int a;
    float b;
}myUnion;

void print_all_members(myUnion *mu){
    printf("a %d\n", mu->a);
    printf("b %f\n", mu->b);
    printf("-----------\n");
}
int main() {
    myUnion mu;
    mu.a = 432;
    print_all_members(&mu);
    mu.b = -20.5;
    print_all_members(&mu);
    return 0;
}

File: ch05/union_undef.zig
const std = @import("std");
const myUnion = union{
    a: i32, 
    b: f64, 
    fn print_all_members(mu: myUnion) void{
        std.debug.print("a {}\n", .{mu.a});
        std.debug.print("b {}\n", .{mu.b});
        std.debug.print(" -----------\n", .{});
    }
};

pub fn main() !void{
    var mu: myUnion = .{.a = 432};
    mu.print_all_members();
    mu = .{.b = -20.5};
    mu.print_all_members();
}